tests/lib/bin/user-tool

```diff
@@ -1,42 +1,55 @@
-# cleans the group
-remove_user_with_group() {
-    REMOVE_USER="$1"
+# Define MYPY as False and use it as a conditional for typing import. Despite
+# this declaration mypy will really treat MYPY as True when type-checking.
+# This is required so that we can import typing on Python 2.x without the
+# typing module installed. For more details see:
+# https://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles
+MYPY = False
+if MYPY:
+    from typing import Text
 
-    if [ -z "$REMOVE_USER" ]; then
-        echo "please provide a username to remove"
-        return 1
-    fi
 
-    if [ -e /var/lib/extrausers/passwd ]; then
-        userdel --extrausers --force "${REMOVE_USER}" || true
-    else
-        userdel --force --remove "${REMOVE_USER}" || true
-        # some systems do not set "USERGROUPS_ENAB yes" so we need to cleanup
+def remove_user_with_group(user_name):
+    # type: (Text) -> None
+    """remove the user and group with the same name, if present."""
+    if os.path.exists("/var/lib/extrausers/passwd"):
+        subprocess.call(["userdel", "--extrausers", "--force", user_name])
+    else:
+        subprocess.call(["userdel", "--force", "--remove", user_name])
+        # Some systems do not set "USERGROUPS_ENAB yes" so we need to cleanup
         # the group manually. Use "-f" (force) when available, older versions
         # do not have it.
-        if groupdel -h | grep force; then
-            groupdel -f "${REMOVE_USER}" || true
-        else
-            groupdel "${REMOVE_USER}" || true
-        fi
-    fi
-    # ensure the ${REMOVE_USER} user really got deleted    
-    not getent passwd "${REMOVE_USER}"
-    not getent group "${REMOVE_USER}"
-}
-
-# main()
-case "$1" in
-    remove-with-group)
-        remove_user_with_group "$2"
-        ;;
-    *)
-        echo "unknown argument $1"
-        echo "look at the source of $0 to see what is supported (sry!)"
-        exit 1
-esac
+        proc = subprocess.Popen(["groupdel", "-h"], stdout=subprocess.PIPE)
+        out, _ = proc.communicate()
+        if b"force" in out:
+            subprocess.call(["groupdel", "-f", user_name])
+        else:
+            subprocess.call(["groupdel", user_name])
+    # Ensure the user user really got deleted
+    if subprocess.call(["getent", "passwd", user_name]) == 0:
+        raise SystemExit("user exists after removal?")
+    if subprocess.call(["getent", "group", user_name]) == 0:
+        raise SystemExit("group exists after removal?")
+
+
+def main():
+    # type: () -> None
+    parser = argparse.ArgumentParser()
+    sub = parser.add_subparsers()
+    cmd = sub.add_parser(
+        "remove-with-group", description="Remove system user and group, if present"
+    )
+    cmd.set_defaults(func=lambda ns: remove_user_with_group(ns.user))
+    cmd.add_argument("user", help="name of the user and group to remove")
+    ns = parser.parse_args()
+    ns.func(ns)
```

Comment:
This is a bit confusing but probably shows I'm just rusty :/

Reply:
This takes the apparmor argument namespace, extracts the `func` argument from it, which is  a function responsible for the command that was selected, runs that function passing the namespace argument back (so that the function can have access to command line arguments). That's all.